db_stress_tool/db_stress_driver.cc

```diff
@@ -58,7 +58,21 @@ void ThreadBody(void* v) {
 
 bool RunStressTest(StressTest* stress) {
   SystemClock* clock = db_stress_env->GetSystemClock().get();
+
   SharedState shared(db_stress_env, stress);
+
+  if (shared.ShouldVerifyAtBeginning() && FLAGS_preserve_unverified_changes) {
+    Status s = InitUnverifiedSubdir(FLAGS_db);
+    if (s.ok() && !FLAGS_expected_values_dir.empty()) {
+      s = InitUnverifiedSubdir(FLAGS_expected_values_dir);
+    }
+    if (!s.ok()) {
+      fprintf(stderr, "Failed to setup unverified state dir: %s\n",
+              s.ToString().c_str());
+      exit(1);
+    }
+  }
```

Comment:
It would be nice to preserve any trace+state file that get removed during `FileExpectedStateManager::Restore()` regardless of `FLAGS_preserve_unverified_changes`'s value. But I am not sure how to make it fit conceptually yet.

Reply:
What's the reason why we need to remove them in FileExpectedStateManager::Restore() other than they takes space if not??

Reply:
Yeah I had the same feeling and tried leaving them there, but it violated an assumption that the any n.{state,trace} that exist can be used to reconstruct LATEST.state: https://github.com/facebook/rocksdb/blob/f79b3d19a77f649536da6321f40e1c0d269bde48/db_stress_tool/db_stress_test_base.cc#L302-L307

If we leave them there through Restore() and delete them later before OperateDb() it is probably fine. There is deletion logic in SaveAtAndAfter() but it isn't enough because non-tracing runs don't call it. So I wasn't sure if there's a clean place to move the deletion logic.